megamek/src/megamek/client/ui/comboBoxes/FontComboBox.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.comboBoxes;
+
+import megamek.client.ui.baseComponents.MMComboBox;
+import megamek.client.ui.displayWrappers.FontDisplay;
+
+import javax.swing.*;
+import java.awt.*;
+
+/**
+ * This is a custom ComboBox that allows you to select a font from your available fonts, each
+ * displayed in that font.
+ */
+public class FontComboBox extends MMComboBox<FontDisplay> {
+    //region Constructors
+    public FontComboBox(final String name) {
+        super(name, FontDisplay.getSortedFontDisplays());
+        setRenderer(new DefaultListCellRenderer() {
+            @Override
+            public Component getListCellRendererComponent(final JList<?> list, final Object value,
+                                                          final int index, final boolean isSelected,
+                                                          final boolean cellHasFocus) {
+                final JLabel label = (JLabel) super.getListCellRendererComponent(list, value, index, isSelected, cellHasFocus);
+                if (value instanceof FontDisplay) {
+                    label.setFont(((FontDisplay) value).getFont());
+                    list.setToolTipText(value.toString());
+                }
+                return this;
```

Comment:
This looks strange because you work on "label" but return "this". I think this does work but it looks off. Maybe return label here. Although I'd even prefer leaving away the "label = " in line 40 and the "label." in l. 42 and keeping the return this.

Reply:
Removed the label and just used setFont, which works perfectly.